### PR TITLE
[JOLINT-51] consent input fields

### PR DESCRIPTION
**Pull request by idamakela**
# Input fields for consent form

- made component for the input elements
- made function to handle the input, only console logs the value object atm
- displays it in the correct area

### CODE REVIEW

File: app/consent-form/_components/form/FormSign.tsx [hunk 1]
```diff
@@ -1,10 +1,75 @@
...
+          <InputField key={item.id} {...item} />
+        ))}
+      </div>
+    </div>
+  )
 }
 
 export default FormSign
+
+/** TODO
```
Comment (DavidDyberg) [inline app/consent-form/_components/form/FormSign.tsx:73]: Remove this comment

**Review by DavidDyberg — CHANGES_REQUESTED** [review]
Remove comment in your code

File: app/consent-form/_components/InputField.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  id,
+  labelValue,
+  name,
+  type,
+  handleForm,
+}: InputFieldType) => {
+  return (
+    <div className='flex gap-2 border-b'>
+      <label htmlFor={htmlFor}>{labelValue}</label>
+      <input type={type} id={id} name={name} onChange={handleForm} />
```
Comment (ell-ska) [inline app/consent-form/_components/InputField.tsx:14]: have a look at the focus state, might look bad😞

File: app/consent-form/_components/form/FormSign.tsx [hunk 3]
```diff
@@ -1,10 +1,75 @@
...
+      htmlFor: 'signature',
+      labelValue: 'Signature:',
+      type: 'text',
+      id: 'signature',
+      name: 'signature',
+      handleForm: (e) => handleForm(e),
+    },
+  ]
+
+  const handleForm = (e: any): void => {
```
Comment (ell-ska) [inline app/consent-form/_components/form/FormSign.tsx:48]: event is not any type

File: app/consent-form/_components/form/FormSign.tsx [hunk 4]
```diff
@@ -1,10 +1,75 @@
...
+    e.preventDefault()
+    const key = e.target.name
+    const value = e.target.value
+    setUserInput((prev) => ({ ...prev, [key]: value }))
+  }
+
+  useEffect(() => {
```
Comment (ell-ska) [inline app/consent-form/_components/form/FormSign.tsx:55]: remove this console log

File: utils/types.ts [hunk 5]
```diff
@@ -44,3 +44,14 @@ export type DropdownProps = {
...
   formater?: (option: string) => string
 }
+
+export type InputFieldType = {
+  htmlFor: string
+  id: string
+  labelValue: string
+  name: string
+  type: string
+  handleForm: (e: any) => void;
```
Comment (ell-ska) [inline utils/types.ts:54]: again event is not any type

**Review by ell-ska — CHANGES_REQUESTED** [review]

**Review by ell-ska — APPROVED** [review]

**Review by DavidDyberg — APPROVED** [review]
Nice!

**Comment by DavidDyberg** [comment]
Dev test.
The input fields looks great!!
Ready for production

**Pull request merged**